app/assets/javascripts/components/ops/log-collection-form-component.js

```diff
@@ -34,22 +46,22 @@ ManageIQ.angular.app.controller('logCollectionFormController', ['$http', '$scope
       miqService.sparkleOn();
 
       var url = vm.logCollectionFormFieldsUrl;
-      $http.get(url + serverId)
+      $http.get(url + vm.recordId)
         .then(getLogCollectionFormData)
         .catch(miqService.handleFailure);
     }
   };
 
   vm.validateClicked = function() {
     miqService.validateWithAjax(vm.saveUrl + '?button=validate&type=' + vm.prefix);
```

Comment:
Validate doesn't work because the backend can't handle `log_protocol: string:Samba` (`string:` prefix added by ng-options).

```diff
diff --git a/app/assets/javascripts/services/miq_service.js b/app/assets/javascripts/services/miq_service.js
index 04df2e397..bcaefc1f0 100644
--- a/app/assets/javascripts/services/miq_service.js
+++ b/app/assets/javascripts/services/miq_service.js
@@ -103,9 +103,9 @@ ManageIQ.angular.app.service('miqService', ['$timeout', '$document', '$q', 'API'
       .catch(options.handleFailure);
   };
 
-  this.validateWithAjax = function(url) {
+  this.validateWithAjax = function(url, model) {
     miqSparkleOn();
-    miqAjaxButton(url, true);
+    miqAjaxButton(url, model || true);
   };
 
   this.validateWithREST = function($event, credType, url, formSubmit) {
```

Should make it possible to pass `vm.logCollectionModel` as a second param to `validateWithAjax`.